src/NuGet.Clients/VisualStudio.Facade/NuGetUIThreadHelper.cs

```diff
@@ -33,28 +33,29 @@ public static JoinableTaskFactory JoinableTaskFactory
         /// Retrieve the CPS enabled JoinableTaskFactory for the current version of Visual Studio.
         /// This overrides the default VsTaskLibraryHelper.ServiceInstance JTF.
         /// </summary>
-        public static void SetJoinableTaskFactoryFromService(IComponentModel componentModel)
+        public static void SetJoinableTaskFactoryFromService(IProjectServiceAccessor projectServiceAccessor)
         {
-            if (componentModel == null)
+            if (projectServiceAccessor == null)
             {
-                throw new ArgumentNullException(nameof(componentModel));
+                throw new ArgumentNullException(nameof(projectServiceAccessor));
             }
 
-            _joinableTaskFactory = new Lazy<JoinableTaskFactory>(() =>
+            if (_joinableTaskFactory == null)
```

Comment:
I think you should add an xml doc comment on this field to state it's always the CPS instance and may be null. It requires reading everything in this class otherwise to understand exactly what it represents and when.